File: 프로그래머스/1/131705. 삼총사/삼총사.cs
using System;

public class Solution {
    public int solution(int[] number) {
        int answer = 0;
        
        for(int i = 0; i < number.Length - 2; i++)
        {
            for(int j = i + 1; j < number.Length - 1; j++)
            {
                int sum = number[i] + number[j];
                for(int k = j + 1; k < number.Length; k++)
                {
                    if((sum + number[k]) == 0)
                        answer++;
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/131705. 삼총사/삼총사.js
function solution(number) 
{
    number = number.sort((a,b) => a - b)
    let count = 0;
    let arr = []
    
    for(let i = 0; i < number.length - 2; i++)
    {
        let temp = number[i]
        for(let j = i + 1; j < number.length - 1; j++)
        {
            for(let k = j + 1; k < number.length; k++)
            {
                number[j] + number[k]
                arr.push(number[j] + number[k])
            }
            
            arr.map((element)=> element + temp == 0 ? count++ : count)
            arr = []
        }
    }
    return count
}